src/Rules/Variables/UnsetRule.php

```diff
@@ -69,6 +69,31 @@ private function canBeUnset(Node $node, Scope $scope): ?IdentifierRuleError
 			}
 
 			return $this->canBeUnset($node->var, $scope);
+		} elseif (
+			$node instanceof Node\Expr\PropertyFetch
+			&& $node->name instanceof Node\Identifier
+		) {
```

Comment:
We have better ways of getting a property reflection from PropertyFetch. Two actually:

1) PropertyReflectionFinder. FoundPropertyReflection will even tell you if it's a native one or not.
2) `Scope::getPropertyReflection()`

Reply:
ahh nice. 

I was already searching how to know about property hooks so I can implement the error for unsetting a hooked property.

will be a followup PR

Reply:
I wrote about property hooks here https://github.com/phpstan/phpstan/discussions/12337